genesis-programs/src/lib.rs

```diff
@@ -163,18 +163,33 @@ mod tests {
     use super::*;
     use std::collections::HashSet;
 
-    #[test]
-    fn test_id_uniqueness() {
-        let mut unique = HashSet::new();
-        let programs = get_programs(OperatingMode::Development);
-        for (program, _start_epoch) in programs {
+    fn do_test_uniqueness(programs: Vec<(Program, Epoch)>) {
+        let mut unique_ids = HashSet::new();
+        let mut unique_names = HashSet::new();
+        let mut prev_start_epoch = GENESIS_EPOCH;
+        for (program, next_start_epoch) in programs {
+            assert!(next_start_epoch >= prev_start_epoch);
             match program {
-                Program::Native((name, id)) => assert!(unique.insert((name, id))),
-                Program::BuiltinLoader((name, id, _)) => assert!(unique.insert((name, id))),
```

Comment:
Well, we must require the uniqueness separately both for `name` and `id`, not for the tuple of `(name, id)`.